Add more examples (#104)
MicrosoftDocs/azure-reference-other: stream-analytics-docs/getarrayelement-azure-stream-analytics.md

 Return type is determined by the array element type and can be any of the [supported types.](data-types-azure-stream-analytics.md)  
  
### Examples  

#### Sample data

```json
[
{
    "DeviceId" : "123",
    "SensorReadings" :
    {
        "Temperature" : 80,
        "Humidity" : 70,
        "CustomSensor": [1,1,0]
    }
},
{
    "DeviceId" : "631",
    "SensorReadings" :
    {
        "Temperature" : 81,
        "Humidity" : 69,
        "CustomSensor": [0,1,0]
    }
}
]
```

> The sample dataset above is an array of 2 records. When used as [local input](https://docs.microsoft.com/en-us/azure/stream-analytics/visual-studio-code-local-run) in a json file, the top level array will be interpreted for the generation of rows / events by Azure Stream Analytics. There is no need to take it into consideration in the query syntax.

Reviewer: ```suggestion
The sample dataset above is an array of two records. When used as [local input](https://docs.microsoft.com/en-us/azure/stream-analytics/visual-studio-code-local-run) in a json file, the top level array is interpreted for the generation of rows/events by Azure Stream Analytics. There is no need to take it into consideration in the query syntax.
```

> At the individual record level, we have 2 properties with different [types](https://docs.microsoft.com/en-us/stream-analytics-query/data-types-azure-stream-analytics). `DeviceId` is of type **nvarchar(max)**, `SensorReadings` is of type **record** (object). [GetType](https://docs.microsoft.com/en-us/stream-analytics-query/gettype-azure-stream-analytics) can be used to determine the type when necessary.

Reviewer: ```suggestion
At the individual record level, there are two properties with different [types](https://docs.microsoft.com/en-us/stream-analytics-query/data-types-azure-stream-analytics). `DeviceId` is of type **nvarchar(max)**, `SensorReadings` is of type **record** (object). [GetType](https://docs.microsoft.com/en-us/stream-analytics-query/gettype-azure-stream-analytics) can be used to determine the type when necessary.
```

> `SensorReadings` has 3 properties, 2 of type **bigint**: `Temperature` and `Humidity`, and finally `CustomSensor` of type **array** (of **bigint**). If this array was in turn more complex (itself containing records or arrays), a combination of [GetArrayElements](https://docs.microsoft.com/en-us/stream-analytics-query/getarrayelements-azure-stream-analytics]) (plural) and [GetRecordPropertyValue](https://docs.microsoft.com/en-us/stream-analytics-query/getrecordpropertyvalue-azure-stream-analytics) could be leveraged.

Reviewer: ```suggestion
`SensorReadings` has three properties: two are of type **bigint**: `Temperature` and `Humidity`, and `CustomSensor` is of type **array** (of **bigint**). If this array was more complex (itself containing records or arrays), a combination of [GetArrayElements](https://docs.microsoft.com/en-us/stream-analytics-query/getarrayelements-azure-stream-analytics]) (plural) and [GetRecordPropertyValue](https://docs.microsoft.com/en-us/stream-analytics-query/getrecordpropertyvalue-azure-stream-analytics) could be used.
```

#### Queries

> 

Reviewer: add an intro to this query

Author: Addressed in [98e4ed4](https://github.com/MicrosoftDocs/azure-reference-other/pull/104/commits/98e4ed4586510f595ac94d76ed0ec9711b42ca79)

```SQL  
SELECT   
	i.DeviceId,
	i.SensorReadings.Temperature,
	i.SensorReadings.Humidity,
	i.SensorReadings.CustomSensor as CustomSensorArray,
	GetArrayElement(i.SensorReadings.CustomSensor,0) AS FirstCustomSensorValue,
  GetArrayElement(i.SensorReadings.CustomSensor,1) AS SecondCustomSensorValue
FROM input i
```

> Will return

Reviewer: ```suggestion
Returns the following output:
```

|DeviceId|Temperature|Humidity|CustomSensorArray|FirstCustomSensorValue|SecondCustomSensorValue|
|---|---|---|---|---|---|
|631|81|69|0,1,0|0|1|
|123|80|70|1,1,0|1|1|

> In order to unfold the array, [CROSS APPLY](apply-azure-stream-analytics.md) can be leveraged:

Reviewer: ```suggestion
Use [CROSS APPLY](apply-azure-stream-analytics.md) to unfold the array:
```

```SQL  
SELECT   
	i.DeviceId,
	CustomerSensorValue.ArrayValue AS CustomerSensorValue
FROM input AS i
CROSS APPLY GetArrayElements(i.SensorReadings.CustomSensor) AS CustomerSensorValue
``` 

> Will return

Reviewer: ```suggestion
Returns the following output:
```

|DeviceId|CustomerSensorValue|
|---|---|
|631|0|
|631|1|
|631|0|
|123|0|
|123|1|
|123|1|

> From there it is straightforward to aggregate the content if necessary:

Reviewer: ```suggestion
From there, you can easily aggregate the content if necessary:
```
